adapters/ddf-adapter/src/test/java/com/connexta/replication/adapters/ddf/rest/DdfRestClientTest.java

```diff
@@ -0,0 +1,205 @@
+/**
+ * Copyright (c) Connexta
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package com.connexta.replication.adapters.ddf.rest;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.connexta.replication.adapters.ddf.MetacardAttribute;
+import com.connexta.replication.adapters.ddf.csw.Constants;
+import com.connexta.replication.adapters.ddf.csw.MetacardMarshaller;
+import com.connexta.replication.data.MetadataImpl;
+import com.connexta.replication.data.ResourceImpl;
+import java.io.ByteArrayInputStream;
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status.Family;
+import javax.ws.rs.core.Response.StatusType;
+import org.apache.commons.io.IOUtils;
+import org.apache.cxf.jaxrs.client.WebClient;
+import org.apache.cxf.jaxrs.ext.multipart.Attachment;
+import org.apache.cxf.jaxrs.ext.multipart.MultipartBody;
+import org.codice.ditto.replication.api.AdapterException;
+import org.codice.ditto.replication.api.data.Metadata;
+import org.codice.ditto.replication.api.data.Resource;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class DdfRestClientTest {
+
+  @Mock WebClient webClient;
+
+  private DdfRestClient client;
+
+  @Before
+  public void setUp() throws Exception {
+    client = new DdfRestClient(webClient);
+  }
+
+  @Test
+  public void postMetadata() {
+    Metadata metadata = getMetadata("123456789");
+    client.post(metadata);
+    verify(webClient).type(MediaType.APPLICATION_XML);
+    verify(webClient).post(MetacardMarshaller.marshal(metadata));
+  }
+
+  @Test
+  public void putMetadata() {
+    Metadata metadata = getMetadata("123456789");
+    client.put(metadata);
+    verify(webClient).type(MediaType.APPLICATION_XML);
+    verify(webClient).path(metadata.getId());
+    verify(webClient).put(MetacardMarshaller.marshal(metadata));
```

Comment:
❓ Should we verify the web clients path is cleared as well?

Reply:
Not sure I understand. The path is verified to be the metadata id above